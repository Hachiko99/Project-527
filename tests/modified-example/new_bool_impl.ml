#use "bool.decls"

let bool_impl : bool -> bool -> bool |>
  { False => True => True
  ; False => False => True
  ; True => True => True
  ; True => False => False} = ?
