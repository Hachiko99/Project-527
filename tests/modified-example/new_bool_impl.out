let bool_impl : bool -> bool -> bool =
  fun (b1:bool) -> fun (b2:bool) -> match b1 with
                                      | True -> b2
                                      | False -> True
;;
